cantabular/table.go

```diff
@@ -54,22 +64,243 @@ func (c *Client) ParseTable(table Table) (*bufio.Reader, error) {
 	return bufio.NewReader(b), nil
 }
 
+// GraphqlJSONToCSV converts a JSON response in r to CSV on w
+// if an error happens, the process is aborted and the error is returned.
+func GraphqlJSONToCSV(r io.Reader, w io.Writer) error {
+	dec := jsonstream.New(r)
+
+	// nullDataset are allowed only if an error message is reported
+	var errData error
+
+	// find starting '{'
+	isStartObj, err := dec.StartObjectComposite()
+	if err != nil {
+		return fmt.Errorf("error decoding start of json object: %w", err)
+	}
+	if !isStartObj {
+		return errors.New("no json object found in response")
+	}
+
```

Comment:
had an experiment, maybe the second version looks a little neater?
```
isStartObj, err := dec.StartObjectComposite()
if err != nil {
	return fmt.Errorf("error decoding start of json object: %w", err)
}
if !isStartObj {
	return errors.New("no json object found in response")
}
```
or
```
if isStartObj, err := dec.StartObjectComposite(); err != nil {
	return fmt.Errorf("error decoding start of json object: %w", err)
} else if !isStartObj {
	return errors.New("no json object found in response")
}
```

Reply:
I agree, I will replace it where needed.